src/plugins/intel_gna/src/gna_graph_compiler.cpp

```diff
@@ -689,18 +696,28 @@ void GNAGraphCompiler::finalizeConvolution2DPrimitive(InferenceEngine::CNNLayerP
                                          convolution._kernel_y,
                                          convolution._kernel_x);
 
-    if (convolution._kernel_x > in_width || convolution._kernel_y > in_height) {
-        THROW_GNA_LAYER_EXCEPTION(layer) << "Kernel dimensions XY (" << convolution._kernel_x << ", "
-                                         << convolution._kernel_y << ")"
-                                         << " are bigger than input dimensions WH (" << in_width << "," << in_height
-                                         << ")";
-    }
+    // Check if kernel width needs to be extended to stride_x.
+    // Needed to avoid regression in comparision to 1D convolution case.
+    const auto effective_kernel_width = std::max(convolution._kernel_x, convolution._stride_x);
+
+    // Check if kernel input needs to be extended for new stride
+    const auto temp_input_for_effective_kernel_with =
+        count_conv2D_input_width_for_expected_output_width(out_width,
+                                                           effective_kernel_width,
+                                                           convolution._stride_x,
+                                                           convolution._padding_x);
+
+    const auto effective_input_width = std::max(in_width, temp_input_for_effective_kernel_with);
+
     const auto inputs = convolution.insData.front().lock();
     const auto outputs = convolution.outData.front();
 
     // have to pad input to let last kernel meets it's corresponding input
-    const auto num_inputs = in_width * in_height * in_channels;
-    uint32_t num_input_padding = ALIGN(num_inputs, 8) - num_inputs;
+    const auto num_inputs = in_batch * effective_input_width * in_height * in_channels;
+
+    // This seems to be redundant for Convolution2D for GNA 3.5 and higher,
```

Comment:
Shouldn't we then enable it just when 3.5 target is chosen? Is it suboptimal otherwise?

Reply:
agreed to not change it.